grype/pkg/package.go

```diff
@@ -56,11 +57,24 @@ func New(p pkg.Package) Package {
 	}
 }
 
-func FromCatalog(catalog *pkg.Catalog) []Package {
+func FromCatalog(catalog *pkg.Catalog, config ProviderConfig) []Package {
 	result := make([]Package, 0, catalog.PackageCount())
+	missingCPEs := false
 	for _, p := range catalog.Sorted() {
+		if len(p.CPEs) == 0 {
+			// For SPDX (or any format, really) we may have no CPEs
+			if config.GenerateMissingCPEs {
+				p.CPEs = cpe.Generate(p)
+			} else {
+				log.Debugf("No CPEs for package: %s", p)
+				missingCPEs = true
+			}
+		}
 		result = append(result, New(p))
 	}
+	if missingCPEs {
+		log.Warnf("Some package(s) are missing CPEs. This may result in missing vulnerabilities. You may autogenerate these using: --add-cpes-if-none")
```

Comment:
nit: 
```suggestion
		log.Warnf("some package(s) are missing CPEs. This may result in missing vulnerabilities. You may autogenerate these using: --add-cpes-if-none")
```